cmd/tsdb/main.go

```diff
@@ -33,12 +33,23 @@ import (
 
 	"github.com/go-kit/kit/log"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/tsdb"
 	"github.com/prometheus/tsdb/chunks"
 	"github.com/prometheus/tsdb/labels"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+func openReadOnlyTSDB(dir string, l log.Logger, r prometheus.Registerer) (db *tsdb.DB, err error) {
```

Comment:
Not sure why those logger and registerer opts if you never fill them (:  I think it proves that just `readOnlyOpts` struct would be enough instead of this function